pkg/capabilities/capabilities.go

```diff
@@ -128,9 +128,23 @@ type BaseCapability interface {
 	Info(ctx context.Context) (CapabilityInfo, error)
 }
 
+type TriggerRegistrationRequest struct {
+	// TriggerID uniquely identifies the trigger by concatenating
+	// the workflow ID and the trigger's index in the spec.
+	TriggerID string
```

Comment:
Are you proposing to replace TriggerID with CapabilityIdx? They aren't totally interchangeable: the TriggerID also contains the workflowID for example. I'm a little reluctant to add CapabilityIdx generally to the RequestMetadata since I don't really understand what it would be used for. The fact that it's used to generate the TriggerID is an implementation detail, not something we should be reverse engineering.